pkg/util/bytesbufpool/bytesbufpool.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bytesbufpool
+
+import (
+	"bytes"
+	"sync"
+)
+
+type BufferPool struct {
+	cap  int // the bytes buffer minimum capacity
+	pool sync.Pool
+}
+
```

Comment:
In general, this package/type needs godocs explaining 

- why you should use this
- when you should use it
- what it does

Reply:
Thanks, simple documentation has been added.